### DISCUSSION TITLE
Issue with arrays in filters?

### DISCUSSION THREAD

**Discussion by p-vanhaver** [spotbugs/spotbugs#3204] [Q&A] [answered]
Hi,

I'm using a filter file in spotbugs-4.8.6 and trying to match a specific method in a class using parameters including an array:
```
<Match>
  <Class name="Foo" />
  <Method name="bar" params="byte[], short, byte" /> 
  <Bug pattern="..." />
</Match>
```

However, I get the following error:
```
java.util.regex.PatternSyntaxException: Unclosed character class near index 9`
\([BSB\).*
         ^
      At java.base/java.util.regex.Pattern.error(Pattern.java:2028)
      At java.base/java.util.regex.Pattern.clazz(Pattern.java:2690)
      At java.base/java.util.regex.Pattern.sequence(Pattern.java:2139)
      At java.base/java.util.regex.Pattern.expr(Pattern.java:2069)
      At java.base/java.util.regex.Pattern.compile(Pattern.java:1783)
      At java.base/java.util.regex.Pattern.<init>(Pattern.java:1430)
      At java.base/java.util.regex.Pattern.compile(Pattern.java:1069)
      At edu.umd.cs.findbugs.filter.NameMatch.<init>(NameMatch.java:89)
      At edu.umd.cs.findbugs.filter.MemberMatcher.<init>(MemberMatcher.java:58)
      At edu.umd.cs.findbugs.filter.MethodMatcher.<init>(MethodMatcher.java:46)
      ...
```

My understanding is that the regex created by `edu.umd.cs.findbugs.filter.SignatureUtil.createMethodSignature()` is missing to escape the `[` because I get `\([BSB\).*` and I should probably get `\(\[BSB\).*`

Am I missing something? Did I incorrectly use the filter?

Thanks for your feedback
--Patrick

**Comment by ThrawnCA** [comment, marked as answer]
Looks like the square bracket should be escaped, yes.

**Comment by JuditKnoll** [comment]
The https://github.com/spotbugs/spotbugs/issues/3203 issue describes the same problem, which got fixed by https://github.com/spotbugs/spotbugs/pull/3205. In the next release this will work properly.
